### PR TITLE
--prove support for --kore 

### PR DESCRIPTION
**Pull request by andreistefanescu**
@bmmoore please review

### CODE REVIEW

**Comment by andreistefanescu** [comment]
@cos please review

File: java-backend/src/main/java/org/kframework/backend/java/compile/KOREtoBackendKIL.java [hunk 1]
```diff
@@ -53,11 +53,13 @@
     private final Module module;
     private final Definition definition;
     private final TermContext context;
+    private final boolean freshRules;
 
-    public KOREtoBackendKIL(Module module, Definition definition, TermContext context) {
+    public KOREtoBackendKIL(Module module, Definition definition, TermContext context, boolean freshRules) {
```
Comment (cos) [inline java-backend/src/main/java/org/kframework/backend/java/compile/KOREtoBackendKIL.java:62]: document freshRule effect

File: java-backend/src/main/java/org/kframework/backend/java/symbolic/InitializeRewriter.java [hunk 2]
```diff
@@ -141,6 +144,37 @@ public RewriterResult execute(K k, Optional<Integer> depth) {
             return Tuple2.apply(res, match(res, rule));
         }
 
+        @Override
+        public List<K> prove(List<Rule> rules) {
```
Comment (cos) [inline java-backend/src/main/java/org/kframework/backend/java/symbolic/InitializeRewriter.java:148]: why does `InitializeRewriter` have a `prove` method?

File: java-backend/src/main/java/org/kframework/backend/java/symbolic/ProofExecutionMode.java [hunk 3]
```diff
@@ -0,0 +1,138 @@
...
+import java.util.Optional;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+import static org.kframework.Collections.*;
+
+/**
+ * Created by dwightguth on 8/26/15.
+ */
+public class ProofExecutionMode implements ExecutionMode<List<K>> {
```
Comment (cos) [inline java-backend/src/main/java/org/kframework/backend/java/symbolic/ProofExecutionMode.java:45]: documentation for the class and its methods -- also, the "Created by" tags are useless considering we have collective code ownership and version control.

File: java-backend/src/main/java/org/kframework/backend/java/symbolic/SymbolicRewriter.java [hunk 4]
```diff
@@ -514,29 +514,33 @@ private boolean addSearchResult(
...
-                    KSequence.getElements(leftKContent).stream().forEach(builder::concatenate);
-                    leftKContent = builder.build();
-                }
-                Term rightKContent = targetTerm.term().getCellContentsByName(CellLabel.K).get(0);
-                Variable rightFrame = KSequence.getFrame(rightKContent);
-                if (rightFrame != null) {
-                    KSequence.Builder builder = KSequence.builder();
-                    KSequence.getElements(rightKContent).stream().forEach(builder::concatenate);
-                    rightKContent = builder.build();
-                }
```
Comment (cos) [inline java-backend/src/main/java/org/kframework/backend/java/symbolic/SymbolicRewriter.java:531]: extract method to remove duplication above

File: java-backend/src/main/java/org/kframework/backend/java/symbolic/SymbolicRewriter.java [hunk 5]
```diff
@@ -514,29 +514,33 @@ private boolean addSearchResult(
...
+                        KSequence.getElements(leftKContent).stream().forEach(builder::concatenate);
+                        leftKContent = builder.build();
+                    }
+                    Term rightKContent = rightKContents.get(0);
+                    Variable rightFrame = KSequence.getFrame(rightKContent);
+                    if (rightFrame != null) {
+                        KSequence.Builder builder = KSequence.builder();
+                        KSequence.getElements(rightKContent).stream().forEach(builder::concatenate);
+                        rightKContent = builder.build();
+                    }
```
Comment (cos) [inline java-backend/src/main/java/org/kframework/backend/java/symbolic/SymbolicRewriter.java]: actually, here

File: kernel/src/main/java/org/kframework/kompile/Kompile.java [hunk 6]
```diff
@@ -226,6 +232,49 @@ private Sentence concretizeSentence(Sentence s, Definition input) {
         return new ConcretizeCells(configInfo, labelInfo, sortInfo, kem).concretize(s);
     }
 
+    public Module parseModule(CompiledDefinition definition, File definitionFile, boolean dropQuote) {
```
Comment (cos) [inline kernel/src/main/java/org/kframework/kompile/Kompile.java:235]: is this method moved from someplace else by any chance?

File: kernel/src/main/java/org/kframework/kore/compile/AddImplicitComputationCell.java [hunk 7]
```diff
@@ -24,7 +26,23 @@ public AddImplicitComputationCell(ConfigurationInfo cfg, LabelInfo labelInfo) {
         this.labelInfo = labelInfo;
     }
 
-    public K apply(K term) {
+    public Sentence apply(Sentence s) {
```
Comment (cos) [inline kernel/src/main/java/org/kframework/kore/compile/AddImplicitComputationCell.java:29]: I left out some cases here. Thanks for covering everything!

File: java-backend/src/main/java/org/kframework/backend/java/symbolic/InitializeRewriter.java (hunk 2 above)
Comment (andreistefanescu) [inline java-backend/src/main/java/org/kframework/backend/java/symbolic/InitializeRewriter.java:148]: technically, `SymbolicRewriterGlue` has the `prove` method. This seems to be the workflow on calling methods from `SymbolicRewriter`.

**Pull request merged**